homeassistant/components/livebox/sensor.py

```diff
@@ -0,0 +1,210 @@
+        self._state = None
+        self._datas = None
+        self._dsl = None
+
+
+class InfoSensor(LiveboxSensor, BinarySensorDevice):
+    """Update Wan Status sensor."""
+
+    device_class = DEVICE_CLASS_CONNECTIVITY
+
+    @property
+    def name(self):
+        """Return name sensor."""
+
+        return TEMPLATE_SENSOR.format("Wan status")
+
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+
+        if self._dsl["WanState"] == "up":
+            return True
+        return False
+
+    @property
+    def unique_id(self):
+        """Return unique_id."""
+
+        return "{}_connectivity".format(self._box_id)
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": "Orange",
+            "via_device": (DOMAIN, self._box_id),
+        }
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+
+        return {
+            "link_type": self._dsl["LinkType"],
+            "link_state": self._dsl["LinkState"],
+            "last_connection_error": self._dsl["LastConnectionError"],
+            "wan_ipaddress": self._dsl["IPAddress"],
+            "wan_ipv6address": self._dsl["IPv6Address"],
+        }
+
+    @Throttle(SCAN_INTERVAL)
+    async def async_update(self):
+        """Fetch status from livebox."""
+
+        self._datas = await self._box.system.get_WANStatus()
+        self._dsl = self._datas["data"]
+
+
+class RXSensor(LiveboxSensor):
+    """Update the Livebox RxSensor."""
+
+    unit_of_measurement = "Mb/s"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+
+        return TEMPLATE_SENSOR.format("download speed")
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+
+        if self._dsl["DownstreamCurrRate"] is None:
+            return None
+        return round(self._dsl["DownstreamCurrRate"] / 1000, 2)
+
+    @property
+    def unique_id(self):
+        """Return unique_id."""
+
+        return "{}_downstream".format(self._box_id)
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": "Orange",
+            "via_device": (DOMAIN, self._box_id),
+        }
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+
+        return {
+            "downstream_maxrate": self._dsl["DownstreamMaxRate"],
+            "downstream_lineattenuation": self._dsl["DownstreamLineAttenuation"],
+            "downstream_noisemargin": self._dsl["DownstreamNoiseMargin"],
+            "downstream_power": self._dsl["DownstreamPower"],
+        }
+
+    @Throttle(SCAN_INTERVAL)
```

Comment:
The throttle doesn't help much since we're not throttling the whole home assistant instance, just this entity instance. It's enough to define the `SCAN_INTERVAL` constant at the module level to set the default update interval.